src/mono/mono/metadata/class-init.c

```diff
@@ -860,7 +862,7 @@ mono_class_create_generic_inst (MonoGenericClass *gclass)
 	if (mono_is_corlib_image (gklass->image) &&
 		(!strcmp (gklass->name, "Vector`1") || !strcmp (gklass->name, "Vector64`1") || !strcmp (gklass->name, "Vector128`1") || !strcmp (gklass->name, "Vector256`1"))) {
 		MonoType *etype = gclass->context.class_inst->type_argv [0];
-		if (mono_type_is_primitive (etype) && etype->type != MONO_TYPE_CHAR && etype->type != MONO_TYPE_BOOLEAN)
+		if (!mini_is_gsharedvt_type (etype))
 			klass->simd_type = 1;
```

Comment:
metadata/ code should not include mini.h, this should be done some other way.

Reply:
I also think this wouldn't completely fix Full AOT on arm64 because `System.Runtime.Intrinsics.WithLower` will be instantiated with `T_GSHAREDVT`, leading to exactly the same problem this was intended to fix; will check now and think about this a little more

Reply:
Maybe it can check klass->simd_type && mini_is_gsharedvt_type (..) in the jit, and fail gsharedvt, using GSHAREDVT_FAILURE ().

Reply:
Tried another approach.